client/chat/channel-header.tsx

```diff
@@ -0,0 +1,252 @@
+  align-items: center;
+  gap: 4px;
+`
+
+const UserCountIcon = styled(MaterialIcon).attrs({ icon: 'groups' })`
+  color: ${colorTextSecondary};
+`
+
+const UserCountText = styled.div`
+  ${caption};
+  color: ${colorTextSecondary};
+`
+
+const StyledIconButton = styled(IconButton)`
+  color: ${colorTextSecondary};
+`
+
+export interface ChannelHeaderProps {
+  basicChannelInfo: ReadonlyDeep<BasicChannelInfo>
+  detailedChannelInfo: ReadonlyDeep<DetailedChannelInfo>
+  joinedChannelInfo: ReadonlyDeep<JoinedChannelInfo>
+  onLeaveChannel: (channelId: SbChannelId) => void
+}
+
+export function ChannelHeader({
+  basicChannelInfo,
+  detailedChannelInfo,
+  joinedChannelInfo,
+  onLeaveChannel,
+}: ChannelHeaderProps) {
+  const { t } = useTranslation()
+
+  const [overflowMenuOpen, openOverflowMenu, closeOverflowMenu] = usePopoverController()
+  const [anchor, anchorX, anchorY] = useAnchorPosition('right', 'bottom')
+
+  const [channelTopicRef, isChannelTopicOverflowing] = useOverflowingElement()
+
+  // TODO(2Pac): Figure out if we can share this with common-message-layout somehow.
+  const parsedChannelTopic = useMemo(() => {
+    if (!joinedChannelInfo.topic) {
+      return undefined
+    }
+
+    const matches = matchLinks(joinedChannelInfo.topic)
+    const sortedMatches = Array.from(matches).sort((a, b) => a.index - b.index)
+    const elements = []
+    let lastIndex = 0
+
+    for (const match of sortedMatches) {
+      // This probably can't happen at this moment, but to ensure we don't get tripped by it in the
+      // future, if this happens we skip the match entirely as it means it overlaps with a previous
+      // match.
+      if (match.index < lastIndex) {
+        continue
+      }
+
+      // Insert preceding text, if any
+      if (match.index > lastIndex) {
+        elements.push(joinedChannelInfo.topic.substring(lastIndex, match.index))
+      }
+
+      if (match.type === 'link') {
+        elements.push(
+          <ExternalLink key={match.index} href={match.text}>
+            {match.text}
+          </ExternalLink>,
+        )
+      } else {
+        assertUnreachable(match.type)
+      }
+
+      lastIndex = match.index + match.text.length
+    }
+
+    // Insert remaining text, if any
+    if (joinedChannelInfo.topic.length > lastIndex) {
+      elements.push(joinedChannelInfo.topic.substring(lastIndex))
+    }
+
+    return elements
+  }, [joinedChannelInfo.topic])
+
+  const onLeaveChannelClick = useStableCallback(() => {
+    onLeaveChannel(basicChannelInfo.id)
+  })
+
+  const actions: React.ReactNode[] = []
+  if (basicChannelInfo.id !== 1 || CAN_LEAVE_SHIELDBATTERY_CHANNEL) {
+    actions.push(
+      <DestructiveMenuItem
+        key='leave-channel'
+        text={t('chat.channelHeader.actionItems.leaveChannel', 'Leave channel')}
+        onClick={onLeaveChannelClick}
+      />,
+    )
+  }
+
+  return (
+    <ChannelHeaderRoot $hasRightPadding={actions.length === 0}>
+      {!basicChannelInfo ? (
```

Comment:
basicChannelInfo is a required prop so I'm not sure why you're doing this check in the first place tbh